src/wallet/bdk_wallet/wallet.rs

```diff
@@ -25,3 +36,191 @@ pub struct Wallet {
+            indexed_tx_graph: indexed_graph.initial_changeset(),
+            network: Some(network),
+        };
+
+        Ok(Wallet {
+            signers: signers_map,
+            chain,
+            indexed_graph,
+            stage: staged,
+            meta_store,
+            network,
+            secp,
+            rpc,
+            wallet_file_path,
+        })
+    }
+}
+
+pub fn initialize_signers_and_update_index(
+    meta_store: &MetaStore,
+    secp: &Secp256k1<All>,
+    index: &mut KeychainTxOutIndex<Keychain>,
+) -> Result<HashMap<Keychain, Arc<SignersContainer>>, DescriptorError> {
+    let mut signers_map = HashMap::new();
+
+    for keychain in [Keychain::External, Keychain::Internal] {
+        let keychain_xpriv = meta_store
+            .master_key
+            .derive_priv(
+                secp,
+                &DerivationPath::from_str(keychain.path().as_str()).unwrap(),
+            )
+            .unwrap();
+
+        // create descriptor strings with checksum
+        let descriptor_without_checksum = format!("wpkh({}/*)", keychain_xpriv);
+
+        let descriptor_with_checksum = format!(
+            "{}#{}",
+            descriptor_without_checksum,
+            calc_checksum(&descriptor_without_checksum).unwrap()
+        );
+
+        // parse descriptor
+        let (descriptor, keymap) =
+            Descriptor::<DescriptorPublicKey>::parse_descriptor(secp, &descriptor_with_checksum)
+                .map_err(DescriptorError::Miniscript)?;
+
+        // mutate index to include corresponding descriptors of a keychain
+        let _ = index
+            .insert_descriptor(keychain, descriptor.clone())
+            .map_err(|e| match keychain {
+                Keychain::External => {
+                    panic!("this is the first descriptor we're inserting")
+                }
+                Keychain::Internal => {
+                    use bdk_chain::keychain::InsertDescriptorError;
+                    match e {
+                        InsertDescriptorError::DescriptorAlreadyAssigned { .. } => {
+                            DescriptorError::ExternalAndInternalAreTheSame
+                        }
+                        InsertDescriptorError::KeychainAlreadyAssigned { .. } => {
+                            unreachable!("this is the first time we're assigning internal")
+                        }
+                    }
+                }
+                _ => unreachable!("we are not assigning other keychains"),
+            })?;
+
+        // build respective signer container
+        let signer_container = Arc::new(SignersContainer::build(keymap, &descriptor, secp));
+        signers_map.insert(keychain, signer_container);
+    }
+
+    Ok(signers_map)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bdk_chain::indexed_tx_graph::Indexer;
+    use bip39::Mnemonic;
+    use bitcoind::tempfile::tempdir;
+
+    #[test]
+    fn test_setup_signers_and_update_index() {
```

Comment:
We should also add another test with Wallet Init. 

Ideally we should test Wallet Init only, and the checks here can be performed in the internal indexes of the new Wallet.

Reply:
To do this you will need a working bitcoind, which can be done easily with the `bitcoind` crate. We have the code in our test framework.

https://github.com/citadel-tech/coinswap/blob/ca3692b9698ab220e0a6f8243a55337667bf390f/tests/test_framework/mod.rs#L105

And you can make a RPC_Config from Bitcoind also using this reference.

https://github.com/citadel-tech/coinswap/blob/ca3692b9698ab220e0a6f8243a55337667bf390f/tests/test_framework/mod.rs#L234-L245